auth/src/main/kotlin/com/hedvig/android/auth/AuthTokenServiceImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.hedvig.android.auth
+
+import com.hedvig.android.auth.storage.AuthTokenStorage
+import com.hedvig.authlib.AccessToken
+import com.hedvig.authlib.AuthRepository
+import com.hedvig.authlib.AuthTokenResult
+import com.hedvig.authlib.RefreshToken
+import com.hedvig.authlib.RefreshTokenGrant
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class AuthTokenServiceImpl(
+  private val authTokenStorage: AuthTokenStorage,
+  private val authRepository: AuthRepository,
+  private val coroutineScope: CoroutineScope,
+) : AuthTokenService {
+
+  @Suppress("NAME_SHADOWING")
+  private val authStatus: StateFlow<AuthStatus?> = authTokenStorage.getTokens()
+    .mapLatest { (accessToken, refreshToken) ->
+      val accessToken = accessToken ?: return@mapLatest AuthStatus.LoggedOut
+      val refreshToken = refreshToken ?: return@mapLatest AuthStatus.LoggedOut
+      AuthStatus.LoggedIn(accessToken, refreshToken)
+    }
+    .stateIn(
+      coroutineScope,
+      SharingStarted.Eagerly,
+      null,
+    )
+
+  override fun getToken(): AccessToken? {
+    return (authStatus.value as? AuthStatus.LoggedIn)?.accessToken
+  }
```

Comment:
Technically this method is just convenience over the `authStatus()` and doing this safe cast, but it's such a common use case that I think it's worth the extra function.